client/src/App.tsx

```diff
@@ -29,30 +26,33 @@ function App(): JSX.Element {
       <ThemeProvider theme={theme}>
         <QueryClientProvider client={queryClient}>
           <ReactQueryDevtools />
-          <UserProvider>
-            <ModalProvider>
-              <BrowserRouter
-                getUserConfirmation={() => {
-                  // intentionally left empty callback to block the default browser prompt.
-                }}
-              >
-                <Grid
-                  className="App"
-                  display="flex"
-                  flexDirection="column"
-                  height="100vh"
-                  width="100vw"
-                  justifyContent="space-between"
+          <ErrorBoundary fallback={GeneralError}>
+            <UserProvider>
+              <ModalProvider>
+                <BrowserRouter
+                  getUserConfirmation={() => {
+                    // intentionally left empty callback to block the default browser prompt.
+                  }}
                 >
-                  {/* move modal to below footer after troubleshooting */}
-                  <Modal />
-                  <Header />
-                  <Main />
-                  <Footer />
-                </Grid>
-              </BrowserRouter>
-            </ModalProvider>
-          </UserProvider>
+                  <Grid
+                    className="App"
+                    display="flex"
+                    flexDirection="column"
+                    height="100vh"
+                    width="100vw"
+                    justifyContent="space-between"
+                  >
+                    <Header />
+                    <ErrorBoundary fallback={NotFound}>
```

Comment:
I think the `fallback` prop should be just like what's in the outer `ErrorBoundary`, being `GeneralError`.  I think I may have confused things when I mentioned that we're not handling 404/page not found error and that we need to reach out to design for this.  I think for this kind of error, instead of using the `ErrorBoundary`, we'd need to add a route that basically shows a 404 page.  I've reached out to design on [#design-ui-ux channel](https://nonprofitcircle.slack.com/archives/C01G4GR05CJ/p1694800626038909) regarding this.